src/Components/Projects/ProjectsForm.js

```diff
@@ -1,48 +1,224 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import Axios from 'axios';
+import { Formik, Form, Field, ErrorMessage } from 'formik';
+import TextField from '@mui/material/TextField';
+import Card from '@mui/material/Card';
+import Button from '@mui/material/Button';
+import CardHeader from '@mui/material/CardHeader';
+import { Alert } from '@mui/material';
+import { CKEditor } from '@ckeditor/ckeditor5-react';
+import ClassicEditor from '@ckeditor/ckeditor5-build-classic';
+import { useDropzone } from 'react-dropzone';
+import Swal from 'sweetalert2';
 import '../FormStyles.css';
+import '../../Styles/ProjectsForm.css';
 
-const ProjectsForm = () => {
+const thumb = {
+  display: 'inline-flex',
+  borderRadius: 2,
+  border: '1px solid #eaeaea',
+  margin: 'auto',
+  width: 'auto',
+  height: 100,
+  padding: 4,
+  boxSizing: 'border-box',
+};
+
+const ProjectsForm = ({ id }) => {
   const [initialValues, setInitialValues] = useState({
-    title: '',
-    description: '',
+    name: id ? id.data.name : '',
+    description: id ? id.data.description : '',
+    dueDate: id ? id.data.dueDate : '',
+    image: id ? id.data.name : null,
   });
+  const [description, setDescription] = useState('');
+  const [files, setFiles] = useState([]);
+  const { getRootProps, getInputProps } = useDropzone({
+    multiple: false,
+    maxFiles: 1,
+    accept: 'image/jpeg, image/png',
+    onDrop: (acceptedFiles) => {
+      setInitialValues({
+        ...initialValues,
+        image: acceptedFiles.map((file) =>
+          Object.assign(file, {
+            preview: URL.createObjectURL(file),
+          }),
+        ),
+      });
 
-  const handleChange = (e) => {
-    if (e.target.name === 'title') {
-      setInitialValues({ ...initialValues, title: e.target.value });
-    }
-    if (e.target.name === 'description') {
-      setInitialValues({ ...initialValues, description: e.target.value });
-    }
+      setFiles(
+        acceptedFiles.map((file) =>
+          Object.assign(file, {
+            preview: URL.createObjectURL(file),
+          }),
+        ),
+      );
+    },
+  });
+
+  const thumbs = files.map((file) => (
+    <div key={file.name} style={thumb}>
+      <div className="thumb-inner">
+        <img className="img" src={file.preview} />
+      </div>
+    </div>
+  ));
+
+  useEffect(
+    () => () => {
+      // Make sure to revoke the data uris to avoid memory leaks
```

Comment:
please delete this comment. The code should be enought documentation to itself. In this case you should create a new function to avoidMemoryLeaks
```
const avoidMemoryLeaks = () => files.forEach((file) => URL.revokeObjectURL(file.preview));
```
then you call it on the useEffect
```